src/sagemaker/local/entities.py

```diff
@@ -76,6 +81,202 @@ def describe(self):
         return response
 
 
+class _LocalTransformJob(object):
+
+    _CREATING = 'Creating'
+    _COMPLETED = 'Completed'
+
+    def __init__(self, transform_job_name, model_name, local_session=None):
+        from sagemaker.local import LocalSession
+        self.local_session = local_session or LocalSession()
+        local_client = self.local_session.sagemaker_client
+
+        self.name = transform_job_name
+        self.model_name = model_name
+
+        # TODO - support SageMaker Models not just local models. This is not
+        # ideal but it may be a good thing to do.
+        self.primary_container = local_client.describe_model(model_name)['PrimaryContainer']
+        self.container = None
+        self.start_time = None
+        self.end_time = None
+        self.batch_strategy = None
+        self.environment = {}
+        self.state = _LocalTransformJob._CREATING
+
+    def start(self, input_data, output_data, transform_resources, **kwargs):
```

Comment:
nit: is it possible to avoid kwargs? It is not something that we agreed yet as a team, so I am fine with the way it is.